pkgs/development/compilers/hip/hipcc.nix

```diff
@@ -0,0 +1,80 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, rocmUpdateScript
+, llvm
+, rocm-device-libs
+, rocm-runtime
+, rocminfo
+, cudatoolkit
+, lsb-release
+, useAMD ? true
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "hipcc-${if useAMD then "amd" else "nvidia"}";
+  version = "5.4.1";
+
+  src = fetchFromGitHub {
+    owner = "ROCm-Developer-Tools";
+    repo = "HIPCC";
+    rev = "rocm-${finalAttrs.version}";
+    hash = "sha256-PEwue4O43MiMkF8UmTeHsmlikBG2V3/nFQLKmtHrRWQ=";
+  };
+
+  postPatch = ''
+    # We are building this separately, so fix the version
+    substituteInPlace CMakeLists.txt \
+      --replace "MAJOR 4 PARENT_SCOPE" "MAJOR ${lib.versions.major finalAttrs.version}" \
+      --replace "MINOR 4 PARENT_SCOPE" "MINOR ${lib.versions.minor finalAttrs.version}" \
+      --replace "PATCH 4 PARENT_SCOPE" "PATCH ${lib.versions.patch finalAttrs.version}"
+
+    # Fix paths
+    substituteInPlace src/hipBin_base.h \
+      --replace "rocm_agent_enumerator_file(roccm_path)" "rocm_agent_enumerator_file(\"${rocminfo}\")"
+
+    substituteInPlace src/{hipBin_amd.h,hipBin_nvidia.h} \
+      --replace "/usr/bin/lsb_release" "${lsb-release}/bin/lsb_release"
+  '' + lib.optionalString useAMD ''
+    substituteInPlace src/hipBin_amd.h \
+      --replace "hsaPath = getRoccmPath()" "hsaPath = \"${rocm-runtime}\"" \
+      --replace "\\\"\" + hipClangPath + \"/clang++" "${stdenv.cc}/bin/clang++" \
+      --replace "complierPath = getRoccmPath()" "complierPath = \"${stdenv.cc}\"" \
+      --replace "hipClangPath + \"/llc" "\"${llvm}/bin/llc" \
+      --replace "amdgcnBitcode = roccmPath" "amdgcnBitcode = \"${rocm-device-libs}/amdgcn/bitcode\"" \
+      --replace "bitcodeFs = roccmPath" "bitcodeFs = \"${rocm-device-libs}/amdgcn/bitcode\"" \
+      --replace "roccmPath+ \"/lib" "\"${rocm-runtime}/lib" \
+      --replace "\" + hipClangPath + \"/../lib/clang/\" +" "${stdenv.cc}/resource-root/lib \";" \
+      --replace "hipClangVersion + \"/lib/linux -lclang_rt.builtins-x86_64 \";" "" \
+      --replace "roccmPath + \"/bin/rocm_agent_enumerator" "\"${rocminfo}/bin/rocm_agent_enumerator"
```

Comment:
With this amount of patching probably using a patch is better and won't slip through on an update.